dwave/system/temperatures.py

```diff
@@ -0,0 +1,488 @@
+
+        T_guess (int, optional, default = 6.1):
+            Determines the range of external fields (h_i) probed for temperature
+            inference; an accurate choice raises the efficiency
+            of the estimator. An inappropriate choice may lead to
+            pathological behaviour. Default is based on D-Wave Advantage 
+            processor temperature and energy scales, and is also suitable for 
+            D-Wave 2000Q processor inference.
+        
+        sampler_params (dict, optional):
+            Any additional non-defaulted sampler parameterization. If num_reads
+            is a key, must be compatible with num_reads argument.
+
+    Returns:
+        float:
+            The effective temperature describing single qubit problems in an
+            external field.
+
+    See also:
+        https://doi.org/10.3389/fict.2016.00023
+    
+    Examples:
+       Draw samples from the default DWaveSampler(), and establish the temperature
+       
+       >>> from dwave.system.temperatures import fast_effective_temperature
+       >>> from dwave.system.samplers import DWaveSampler
+       >>> sampler = DWaveSampler() # doctest: +SKIP
+       >>> T = fast_effective_temperature(sampler)     # doctest: +SKIP
+       >>> print('Effective temperature at freeze-out is',T)     # doctest: +SKIP
+    '''
+    
+    if sampler == None:
+        from dwave.system.samplers import DWaveSampler
+        sampler = DWaveSampler()
+    h_range = [-2/T_guess, 2/T_guess]
+    if hasattr(sampler,'properties') and 'h_range' in sampler.properties:
+        warn_user = False
+        if h_range[0] < sampler.properties['h_range'][0]:
+           h_range[0] = sampler.properties['h_range'][0]
+           warn_user = True
+        if h_range[1] > sampler.properties['h_range'][1]:
+            h_range[1] = sampler.properties['h_range'][1]
+            warn_user = True
+        if warn_user:
+            warnings.warn(
+                'T_guess is small (relative to programmable h_range). '
+                'Maximum h_range is employed, but this may be '
+                'statistically inefficient.')
+            
+    prng = np.random.RandomState(seed)
+    h_values = h_range[0] + (h_range[1]-h_range[0])*prng.rand(len(sampler.nodelist))
+    bqm = dimod.BinaryQuadraticModel.from_ising({var: h_values[idx] for idx,var in enumerate(sampler.nodelist)}, {})
+    if sampler_params == None:
+        sampler_params = {}
+    if num_reads == None:
+        #Default is 100, makes efficient use of QPU access time:
+        if 'num_reads' not in sampler_params:
+            sampler_params['num_reads'] = 100 
+    elif ('num_reads' in sampler_params
+        and sampler_params['num_reads'] != num_reads):
+        raise ValueError("sampler_params['num_reads'] != num_reads, "
+                         "incompatible input arguments.")
+    else:
+        sampler_params['num_reads'] = num_reads
+    sampler_params['auto_scale'] = False
```

Comment:
Before doing this should you not check whether the user has set this parameter in ``sampler_params``? Otherwise you're silently overriding the user's configuration

Reply:
I copied, and then added any missing keys, should be fine I assume.